### PR TITLE
#62188 OAuth 2.0 - Support Authorization code grant (PKCE)

### PR DESCRIPTION
**Pull request by ctthang**

### CODE REVIEW

File: src/CSharp/AspnetWebMvc/AspnetWebMvc/Web.config [hunk 1]
```diff
@@ -1,41 +1,43 @@
...
+    <add key="IdentifyOauth2:CodeVerifier" value="nX8nXTvRo5pAA1GedqTYF7yLykImlbUs9gEnkm-6ixI" />  <!--URL safe unique identifier-->
```
Comment (PhuongLe) [inline src/CSharp/AspnetWebMvc/AspnetWebMvc/Web.config]: This value should be auto-generated inside the source code. Configuration setting added to enable/disable it only

File: src/CSharp/AspnetWebMvc/AspnetWebMvc/Web.config [hunk 2]
```diff
@@ -1,41 +1,43 @@
...
+    <add key="IdentifyOauth2:CodeChallengeMethod" value="plain" />    <!--S256 or plain-->
```
Comment (PhuongLe) [inline src/CSharp/AspnetWebMvc/AspnetWebMvc/Web.config]: Sha256? 
Btw, you can provide 3 options: SHA256, Plain or None, in which, none means PKCE is disabled.

File: src/CSharp/AspnetWebMvc/AspnetWebMvc/Web.config (hunk 1 above)
Comment (ctthang) [reply src/CSharp/AspnetWebMvc/AspnetWebMvc/Web.config]: Yes, I remove the setting for "code verifier" and generate it randomly for each session.

File: src/CSharp/AspnetWebMvc/AspnetWebMvc/Web.config (hunk 2 above)
Comment (ctthang) [reply src/CSharp/AspnetWebMvc/AspnetWebMvc/Web.config]: Add "none" option.
the values should be "S256" and "plain" as the spec https://tools.ietf.org/html/rfc7636#section-4.2

**Pull request closed without merging**